File: sdk/client/javascript/openapi-test/test/klaytnDebug/blockchainInspection/DebugGetModifiedAccountsByNumber.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_getModifiedAccountsByNumber API', () => {
    test('should return debug_getModifiedAccountsByNumber', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data.jsonrpc).toBe("2.0");
            expect(data.result).toBeDefined()
            done();
        };

        const startBlockNum = 171904
        const endBlockNum = 172160

        sdk.debug.getModifiedAccountsByNumber(startBlockNum, {endBlockNum}, callbackOne);
    });
});

File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/debug/blockchainInspection/DebugGetModifiedAccountsByNumberTest.java
package opensdk.sdk.apis.debug.blockchainInspection;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugGetModifiedAccountsByNumberResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugGetModifiedAccountsByNumberTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.TEST_URL);

    @Test
    @DisplayName("RPC debug_getModifiedAccountsByNumber")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        int startBlockNum = 171904;
        int endBlockNum = 172160;
        DebugGetModifiedAccountsByNumberResponse response = sdk.debug.getModifiedAccountsByNumber(startBlockNum, endBlockNum).send();
        assertNotNull(response);
        assertNull(response.getError());
    }
}
